From 04247ae2a1141518d3d104e37be7b1a980a49c32 Mon Sep 17 00:00:00 2001
From: Emmanuel Jillela <emmanuel.jillela@intel.com>
Date: Wed, 3 Feb 2016 16:48:52 -0700
Subject: [PATCH 257/441] Title: Adding the BBU device to NETIP MMIO info API
 Issue: Jira CPM-252 Resolution: netss_device_get_info() API can be used to
 query BBU controllers MMIO address and size.

---
 arch/x86/NetIP_SubSystem/netip_subsystem_lld.c | 5 +++++
 include/linux/netip_subsystem.h                | 3 +++
 2 files changed, 8 insertions(+)

--- a/arch/x86/NetIP_SubSystem/netip_subsystem_lld.c
+++ b/arch/x86/NetIP_SubSystem/netip_subsystem_lld.c
@@ -198,6 +198,11 @@ int netss_device_get_info(netss_dev_t su
          mmio->size = NETSS_DEV_CLK_CTRL_MMIO_SIZE;
          ret = 0;
       break;
+      case NETSS_DEV_BBU_CTRLR:
+         mmio->base = net_ip_mmios.region1_base + NETSS_DEV_BBU_CTRLR_MMIO_OFFSET;
+         mmio->size = NETSS_DEV_BBU_CTRLR_MMIO_SIZE;
+         ret = 0;
+      break;
       default:
          ret = -1;
    }
--- a/include/linux/netip_subsystem.h
+++ b/include/linux/netip_subsystem.h
@@ -57,6 +57,8 @@
 #define NETSS_DEV_GBE_MMIO_SIZE (0x6000)
 #define NETSS_DEV_CLK_CTRL_MMIO_OFFSET (0xD0000)
 #define NETSS_DEV_CLK_CTRL_MMIO_SIZE (0x10000)
+#define NETSS_DEV_BBU_CTRLR_MMIO_OFFSET (0x160000)
+#define NETSS_DEV_BBU_CTRLR_MMIO_SIZE   (0x20000)  /*128 KB */
 
 #define NETSS_INTERUPT_ARM11_MAX (64)  /*ARM11 is a bundle of 64 interrupts */
 
@@ -70,6 +72,7 @@ typedef enum {
    NETSS_DEV_MOCA,
    NETSS_DEV_GBE,
    NETSS_DEV_CLK_CTRL,
+   NETSS_DEV_BBU_CTRLR,
    NETSS_DEV_MAX
 }netss_dev_t;
 
